packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -70,7 +70,7 @@ export class ProfilesCache {
 
     public async getProfileInfo(): Promise<zowe.imperative.ProfileInfo> {
         const mProfileInfo = new zowe.imperative.ProfileInfo("zowe");
-        await mProfileInfo.readProfilesFromDisk({ homeDir: getZoweDir(), projectDir: this.cwd });
+        await mProfileInfo.readProfilesFromDisk({ homeDir: getZoweDir(), projectDir: this.cwd ?? "" });
```

Comment:
should there be a check for if this.cwd is the OS's home directory?

Reply:
something similar to this fashion `const homedir = require('os').homedir();`

Reply:
The scenario I have tested so far with this PR is when there is no workspace open in VS Code, to ensure that project config is not loaded. The scenario you mentioned seems different - if the project directory is the same as my home directory, we don't want the config files in the home directory to get loaded twice. Is that right? No objections to testing that scenario as well and making sure this PR handles it correctly, just want to make sure we're on the same page 🙂

Reply:
Added unit tests to verify that loading profiles from home directory and project directory behaves correctly.